Dockerfile

```diff
@@ -8,8 +8,12 @@ WORKDIR /app
 # amvn will rebuild+restart the server when .java files change
 # Logged in users (not guests) will reconnected by the UI within ~12s.
 
-# consider removing clean for slightly faster comilation
-ENTRYPOINT amvn clean package exec:java --watch
+# "amvn clean" is left out for faster iterative compilation.
+# Add it back locally if necessary for your workflow, but
+# rebuilding your docker image should accomplish the same goal.
+# This should be changed also if docker is used for deployment.
+# ENTRYPOINT amvn clean package exec:java --watch
+ENTRYPOINT amvn package exec:java --watch
```

Comment:
1. @invaderb performance improvement iirc 1-2 seconds
2. why `clean`? Because `mvn` compiles `.java` files into bytecode in the output directory, but never cleans the output directory. Imagine refactoring a few classes and moving them to other places. If you don't `clean`, then code referencing the old classes will continue to work until `clean` is executed - [see stackoverflow why run mvn clean](https://stackoverflow.com/questions/4662452/in-maven-why-run-mvn-clean)
4. This gets **even worse** when switching between branches, that have different feature sets in different stages.
5. `rebuilding your docker image should accomplish the same goal` - probably not, because `docker-compose.yml` for `tak-server` mounts volume `- ./tak-server:/app` (i.e. it mounts the entire folder ./tak-server into the container. Thus all output that lands in that folder - which likely includes the generated bytecode - will persist container restarts or rebuilds)

I strongly recommend sticking with `clean` for the following reason: It can lead to very hard to debug issues while developing ("y u no work?"/"doesn't work on my machine, but everywhere else" 🙀 ). So if you fancy faster builds and know what you're doing, remove `clean` on your machine - but for anyone not actively thinking about this, it's a trap to fall into.

Reply:
@devp

Reply:
If you agree, feel free to close the PR.

Reply:
Brilliant, thanks for the great explanation for someone with less comfort with Java. Will happily decline.